scripts/devops_tasks/test_regression.py

```diff
@@ -0,0 +1,364 @@
+                self.context.package_root_path,
+            )
+            # Install pre-built whl for current package
+            install_package_from_whl(
+                self.context.package_name,
+                self.context.whl_directory,
+                self.context.temp_path,
+                self.context.python_symlink,
+            )
+            # install package to be tested and run pytest
+            self._execute_test(dep_pkg_path)
+        finally:
+            # deactivate virtual env and revert repo
+            self.context.deactivate(dep_pkg_path)
+
+
+    def _execute_test(self, dep_pkg_path):
+        # install dependent package from source
+        self._install_packages(dep_pkg_path, self.context.package_name)
+        logging.info("Running test for {}".format(dep_pkg_path))
+        pkg_test_dir = self._get_package_test_dir(dep_pkg_path)
+        run_check_call(
+            [
+                self.context.python_symlink,
+                "-m",
+                "pytest",
+                "--verbose",
+                "-m",
+                COSMOS_TEST_ARG,
+                pkg_test_dir,
+            ],
+            self.context.temp_path,
+        )
+
+    def _get_package_test_dir(self, pkg_root_path):
+        # Returns path to test or tests folder within package root directory.
+        paths = glob.glob(os.path.join(pkg_root_path, "test*"))
+        if paths is None:
+            logging.error("'test' folder is not found in {}".format(pkg_root_path))
+            sys.exit(1)
+        return paths[0]
+
+    def _install_packages(self, dependent_pkg_path, pkg_to_exclude):
+        python_executable = self.context.python_symlink
+        working_dir = self.context.package_root_path
+        temp_dir = self.context.temp_path
+
+        # install dev requirement but skip already installed package which is being tested
+        filtered_dev_req_path = filter_dev_requirements(
+            dependent_pkg_path, [pkg_to_exclude,], dependent_pkg_path
+        )
+        logging.info(
+            "Installing filtered dev requirements from {}".format(filtered_dev_req_path)
+        )
+        run_check_call(
+            [python_executable, "-m", "pip", "install", "-r", filtered_dev_req_path],
+            dependent_pkg_path,
+        )
+        # install dependent package which is being verified
+        run_check_call(
+            [python_executable, "-m", "pip", "install", dependent_pkg_path], temp_dir
+        )
+
+
+# This method identifies package dependency map for all packages in azure sdk
+def find_package_dependency(glob_string, repo_root_dir):
+    package_paths = process_glob_string(
+        glob_string, repo_root_dir, "", OmmitType.Regression
+    )
+    dependency_map = {}
+    for pkg_root in package_paths:
+        dependent_pkg_name, _, _, requires = parse_setup(pkg_root)
+        # todo: This should be removed once issue in storage file share test is resolved
+        if "storage" in dependent_pkg_name:
```

Comment:
Don't use "storage" here, use the storagefileshare one you have defined up above.